src/lib.rs

```diff
@@ -5,6 +5,7 @@ use std::fmt::Debug;
 use std::path::{Path, PathBuf};
 use std::rc::Rc;
 use std::hash::{Hash, Hasher};
+use std::borrow::Borrow;
```

Comment:
I had to change all the code that used RefCell and Rc in this file, I don't really know why, and it annoys me. But this made the PR bigger than it should have been.